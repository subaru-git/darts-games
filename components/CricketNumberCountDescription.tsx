import {
  Button,
  Collapse,
  useDisclosure,
  Text,
  UnorderedList,
  ListItem,
} from "@chakra-ui/react";
import React, { FC } from "react";
import { GrCircleInformation } from "react-icons/gr";
import YouTube from "react-youtube";

const CricketNumberCountDescription: FC = () => {
  const { isOpen, onToggle } = useDisclosure();
  return (
    <>
      <Button
        leftIcon={<GrCircleInformation />}
        variant="ghost"
        size="sm"
        onClick={onToggle}
      >
        What is this game?
      </Button>
      <Collapse in={isOpen}>
        <Text>
          This game was designed by&#39;Kikuchi Yamaguchi.&#39; <br />
          The game rule is:
        </Text>
        <UnorderedList py={2}>
          <ListItem>Setting target count (recommended 10 times)</ListItem>
          <ListItem>
            Shoot for each cricket numbers from 20 to 15 and Bull until the
            target count of times, like Cricket.
          </ListItem>
        </UnorderedList>
        <Text>For more information, please watch this video.</Text>
        <YouTube videoId="RFl52NNvWhw" />
      </Collapse>
    </>
  );
};

export default CricketNumberCountDescription;
