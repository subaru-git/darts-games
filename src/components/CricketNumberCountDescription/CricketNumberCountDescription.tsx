import React, { FC } from 'react'
import { Button, Collapse, useDisclosure, Text, UnorderedList, ListItem } from '@chakra-ui/react'
import { GrCircleInformation } from 'react-icons/gr'
import YouTube from 'react-youtube'
import useLocale from '@/hooks/locale'

const CricketNumberCountDescription: FC = () => {
  const { isOpen, onToggle } = useDisclosure()
  const { t } = useLocale()
  return (
    <>
      <Button leftIcon={<GrCircleInformation />} variant='ghost' size='sm' onClick={onToggle}>
        What is this game?
      </Button>
      <Collapse in={isOpen}>
        <Text whiteSpace='pre-wrap'>{t.CRICKET_NUMBER_COUNT_DESCRIPTION}</Text>
        <UnorderedList py={2}>
          <ListItem>{t.CRICKET_NUMBER_COUNT_RULE_STEP1}</ListItem>
          <ListItem>{t.CRICKET_NUMBER_COUNT_RULE_STEP2}</ListItem>
        </UnorderedList>
        <Text>{t.CRICKET_NUMBER_COUNT_RULE_MORE}</Text>
        <YouTube videoId='RFl52NNvWhw' />
      </Collapse>
    </>
  )
}

export default CricketNumberCountDescription
