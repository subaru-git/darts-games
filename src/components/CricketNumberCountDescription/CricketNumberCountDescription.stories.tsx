/* eslint-disable import/named */
import React from 'react'
import { ComponentMeta, ComponentStory } from '@storybook/react'
import CricketNumberCountDescription from './CricketNumberCountDescription'

export default {
  title: 'Components/CricketNumberCountDescription',
  component: CricketNumberCountDescription,
} as ComponentMeta<typeof CricketNumberCountDescription>

const Template: ComponentStory<typeof CricketNumberCountDescription> = (args) => (
  <CricketNumberCountDescription {...args} />
)

export const Default = Template.bind({})
Default.args = {}
