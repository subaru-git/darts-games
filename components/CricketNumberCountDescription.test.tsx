import { render, screen } from '@testing-library/react'
import CricketNumberCountDescription from './CricketNumberCountDescription'

test('should rendering', () => {
  const { container } = render(<CricketNumberCountDescription />)
  expect(screen.getByText('What is this game?')).not.toBeNull()
  expect(container).toMatchSnapshot()
})
