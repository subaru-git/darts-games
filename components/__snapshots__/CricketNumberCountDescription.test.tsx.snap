// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should rendering 1`] = `
<div>
  <button
    class="chakra-button css-0"
    type="button"
  >
    <span
      class="chakra-button__icon css-0"
    >
      <svg
        aria-hidden="true"
        fill="currentColor"
        focusable="false"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12,22 C17.5228475,22 22,17.5228475 22,12 C22,6.4771525 17.5228475,2 12,2 C6.4771525,2 2,6.4771525 2,12 C2,17.5228475 6.4771525,22 12,22 Z M12,10 L12,18 M12,6 L12,8"
          fill="none"
          stroke="#000"
          stroke-width="2"
        />
      </svg>
    </span>
    What is this game?
  </button>
  <div
    class="chakra-collapse"
    style="overflow: hidden; display: none; opacity: 0; height: 0px;"
  >
    <p
      class="chakra-text css-0"
    >
      This game was designed by 'Kikuchi Yamaguchi.'
The game rule is:
    </p>
    <ul
      class="css-0"
      role="list"
    >
      <li
        class="css-0"
      >
        Setting target count (recommended 10 times)
      </li>
      <li
        class="css-0"
      >
        Shoot for each cricket numbers from 20 to 15 and Bull until the target count of times, like Cricket.
      </li>
    </ul>
    <p
      class="chakra-text css-0"
    >
      For more information, please watch this video.
    </p>
    <div
      class=""
    >
      <div
        class=""
        id=""
      />
    </div>
  </div>
</div>
`;
